semanticcpg/src/main/scala/io/shiftleft/semanticcpg/language/types/structure/MethodReturn.scala

```diff
@@ -33,7 +33,5 @@ class MethodReturn(val wrapped: NodeSteps[nodes.MethodReturn]) extends AnyVal {
     new NodeSteps(raw.out(EdgeTypes.EVAL_TYPE).cast[nodes.Type])
 
   def toReturn: NodeSteps[nodes.Return] =
-    new NodeSteps(raw.flatMap { mr =>
-      __(mr.toReturn.toSeq: _*)
-    })
+    new NodeSteps(raw.map(_.toReturn).collect { case Some(ret) => ret })
```

Comment:
Just `.flatMap(_.toReturn)` should be enough.

Reply:
unfortunately that doesn't compile, because `_.toReturn` returns an `Option[nodes.Return]`, and `flatMap` needs a `Steps` instance. 

Btw once we swap to Odb Traversal the flatMap you proposed should work (since Odb Traversal is a scala collection): 
```
List(1,2,3).flatMap{ _ match { case 1|2 => None; case 3 => Some(4)} }
val res2: List[Int] = List(4)
```